### PR TITLE
Clarifications of code ownership in /ledger and /ledger-api

### PR DESCRIPTION
**Pull request by gerolf-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by gerolf-da** [comment]
@meiersi-da, @mziolekda, @miklos-da: I attempted to clarify the ownership of various parts of `/ledger` and `/ledger-api`. Could you please review to check whether you have the same understanding?

File: CODEOWNERS [hunk 1]
```diff
@@ -30,25 +30,31 @@ NOTICES     @garyverhaegen-da @aherrmann-da @cocreature @dasormeter
...
+/ledger/ledger-api-test-tool/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-configuration/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-offset/              @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/participant-state-index/    @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/sandbox/                    @digital-asset/kv-participant @digital-asset/kv-committer
+
+# KV Committer
 /ledger/ledger-on-memory/           @digital-asset/kv-committer
 /ledger/ledger-on-sql/              @digital-asset/kv-committer
 /ledger/participant-state/kvutils/  @digital-asset/kv-committer
```
Comment (meiersi-da) [inline CODEOWNERS:60]:
```suggestion
/ledger/participant-state/kvutils/  @digital-asset/kv-committer

# Jointly owned persistence schema and wire format
/ledger/participant-state/kvutils/src/main/protobuf  @meiersi-da @digital-asset/kv-committer

```

**Review by meiersi-da — COMMENTED** [review]
looks good overall. Added myself to two further paths.

File: CODEOWNERS [hunk 2]
```diff
@@ -30,25 +30,31 @@ NOTICES     @garyverhaegen-da @aherrmann-da @cocreature @dasormeter
...
 
+# Owned by KV Participant with KV Committer added for notifications
+/ledger/ledger-api-common/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-domain/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-health/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-test-tool/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-configuration/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-offset/              @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/participant-state-index/    @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/sandbox/                    @digital-asset/kv-participant @digital-asset/kv-committer
```
[the remark this answers is not in the document]
Comment (gerolf-da) [reply CODEOWNERS:55]: Would you prefer to be added to the schema definition directly (i.e. the migration files)?

[the remark this answers is not in the document]
Comment (meiersi-da) [reply CODEOWNERS:55]: I was thinking about that. However, given the intricacy of the codebase, I'd also like to keep up to date on the implementation of the indexer, in-memory fan-out, mutable state cache etc. Let's leave the line like this for now, and I'll create a PR if I get too much traffic in my inbox.

File: CODEOWNERS [hunk 3]
```diff
@@ -30,29 +30,43 @@ NOTICES     @garyverhaegen-da @aherrmann-da @cocreature @dasormeter
...
-/ledger/participant-state/          @digital-asset/kv-committer @digital-asset/kv-participant
-/ledger/participant-state-index/    @digital-asset/kv-committer @digital-asset/kv-participant
-/ledger/participant-state-metrics/  @digital-asset/kv-committer @digital-asset/kv-participant
-/ledger/sandbox/                    @digital-asset/kv-committer @digital-asset/kv-participant
+# Interfaces that are jointly owned
+/ledger-api/grpc-definitions/       @meiersi-da @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/participant-state/          @meiersi-da @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/participant-state-metrics/  @meiersi-da @digital-asset/kv-participant @digital-asset/kv-committer
 
+# Owned by KV Participant with KV Committer added for notifications
```
Comment (miklos-da) [inline CODEOWNERS:47]: Some of the below need to be co-owned as well, such as the below:
* `/ledger/ledger-api-test-tool` (this regularly needs to be modified based on participant state API _implementation_)
* `/ledger/ledger-configuration/`, `/ledger/ledger-offset` (part of participant state API)
We might want to reconsider the module structure for the latter, and e.g. move all the participant state API related modules into a separate subdirectory (such as `/ledger/participant-state-api/...`).

**Review by cocreature — APPROVED** [review]

File: CODEOWNERS [hunk 4]
```diff
@@ -30,29 +30,45 @@ NOTICES     @garyverhaegen-da @aherrmann-da @cocreature @dasormeter
...
+/ledger/ledger-api-common/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-domain/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-health/          @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-api-test-tool/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-configuration/       @digital-asset/kv-participant @digital-asset/kv-committer
+/ledger/ledger-offset/              @digital-asset/kv-participant @digital-asset/kv-committer
```
Comment (miklos-da) [inline CODEOWNERS:55]: These are duplicate entries.

**Pull request merged**